base/workflow/inst/batch_runs.R

```diff
@@ -0,0 +1,431 @@
+con <- bety$con
+
+## Find name of Machine R is running on
+machid_rxp <- "^--machine_id="
+if (any(grepl(machid_rxp, argv))) {
+  machid_raw <- grep(machid_rxp, argv, value = TRUE)
+  mach_id <- as.numeric(gsub(machid_rxp, "", machid_raw))
+  message("Using specified machine ID: ", mach_id)
+} else {
+  mach_name <- Sys.info()[[4]]
+  message("Auto-detected machine name: ", mach_name)
+  ## mach_name <- "docker"
+  mach_id <- tbl(bety, "machines") %>%
+    filter(hostname == !!mach_name) %>%
+    pull(id)
+}
+  
+## Find Models
+#devtools::install_github("pecanproject/pecan", subdir = "api")
+model_df <- tbl(bety, "dbfiles") %>%
+  filter(machine_id == !!mach_id) %>%
+  filter(container_type == "Model") %>%
+  left_join(tbl(bety, "models"), c("container_id" = "id")) %>%
+  select(model_id = container_id, model_name, revision,
+         file_name, file_path, dbfile_id = id, ) %>%
+  collect() %>%
+  mutate(exists = file.exists(file.path(file_path, file_name)))
+
+message("Found the following models on the machine:")
+print(model_df)
+
+if (!all(model_df$exists)) {
+  message("WARNING: The following models are registered on the machine ",
+          "but their files do not exist:")
+  model_df %>%
+    filter(!exists) %>%
+    print()
+
+  model_df <- model_df %>%
+    filter(exists)
+}
+
+models <- model_df[["model_id"]]
+
+## Find Sites
+## Site with no inputs from any machines that is part of Ameriflux site group and Fluxnet Site group
+site_id_noinput <- tbl(bety, "sites") %>%
+  anti_join(tbl(bety, "inputs")) %>%
+  inner_join(tbl(bety, "sitegroups_sites") %>%
+               filter(sitegroup_id == 1),
+             by = c("id" = "site_id")) %>%
+  dplyr::select("id.x", "notes", "sitename") %>%
+  dplyr::filter(grepl("TOWER_BEGAN", notes)) %>%
+  collect() %>%
+  dplyr::mutate(
+    # Grab years from string within the notes
+    start_year = substring(stringr::str_extract(notes,pattern = ("(?<=TOWER_BEGAN = ).*(?=  TOWER_END)")),1,4),
+    #Empty tower end in the notes means that it goes until present day so if empty enter curent year.
+    end_year = dplyr::if_else(
+      substring(stringr::str_extract(notes,pattern = ("(?<=TOWER_END = ).*(?=)")),1,4) == "",
+      as.character(lubridate::year(Sys.Date())),
+      substring(stringr::str_extract(notes,pattern = ("(?<=TOWER_END = ).*(?=)")),1,4)
+    ),
+    #Check if startdate year is within the inerval of that is given
+    in_date = data.table::between(as.numeric(lubridate::year(startdate)),as.numeric(start_year),as.numeric(end_year))
+  ) %>%
+  dplyr::filter(
+    in_date,
+    as.numeric(end_year) - as.numeric(start_year) > 1
+  ) %>%
+  mutate(sitename = gsub(" ", "_", sitename)) %>%
+  rename(site_id = id.x)
+
+message("Running tests at ", nrow(site_id_noinput), " sites:")
+print(site_id_noinput)
+
+site_id <- site_id_noinput$site_id
+site_name <- gsub(" ", "_", site_id_noinput$sitename)
+
+
+
+#Create permutations of arg combinations
+options(scipen = 999)
+run_table <- expand.grid(
```

Comment:
I don't really know how expand.grid works, but it's unclear to me from the syntax below which things are being expanded in all possible permutations and which are held fixed (and if they're fixed, how you designate which attribute they line up with). So for global models we want to permute over models x sites, but not all models make sense to run at all sites (LINKAGES only for east temperate, crop models only on crop sites, etc). Similarly, we probably don't want to permute all sites to run all possible met combinations, since AmerifluxLBL won't exist for arbitrary sites on the globe and as we expand the list of met products we want to test we really just need a handful of runs for the gridded products to verify they are working. And the start/end dates are specific to a site x met combination -- how does this function know that (and, for example, that they're not model attributes)? Oh, and PFTs are also a model x site specific setting, where do those come in?  And where do you specify other inputs (IC, soils, etc)? All in all, to me it just seems simpler to have a table of what runs you want to do at what sites with what inputs than to try to write code that creates all possible permutations

Reply:
`expand.grid` generates a full permutation of everything in it. So this is a full factorial combination of all arguments. However, some of them are scalar, so they are fixed.

This structure was in the original code, and I took as my responsibility to just get it to run and produce some kind of output. I can revise the structure of this script to read in a table of runs, but the first draft of that table will be pretty limited because I don't have time to figure out which combination of sites, PFTs, models, and other settings makes sense.

Reply:
That's fair. I'm also fine with pulling this PR with expand.grid still there and without deeper documentation under the promise that those things are coming soon. Happy to help figuring out what model, site, met, date combinations make sense (e.g. in a Google Sheet)